.gitignore

```diff
@@ -16,3 +16,6 @@ workspace.code-workspace
 *.pro.shared
 
 .vscode
+*.log
+null.d
```

Comment:
What is null.d?

Reply:
it's generated by the makefile extension.

Reply:
I can't find any reference to null.d on the help page. It is strange that they choose NUL.D is a reserved filename on Windows. https://answers.microsoft.com/en-us/windows/forum/all/unable-to-rename-a-folder-to-com-errorthe/ff48c793-ab97-4881-952a-df195321c11c

Reply:
yeah. i haven't been able to figure it out. i just know it gets generated when they generate the list of make options. and it's not consistent either.

Reply:
Remove from PR then, and complain to microsoft :)